### ISSUE TITLE
[DataGrid] Cannot use arrow keys to move the focus of a cell that contains a Button

### ISSUE THREAD

**Issue by TaisonTsukada** [mui/mui-x#4068]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

If a Button is included in a DataGrid cell, it is not possible to control the focus of the cell with the arrow keys, and the screen scrolls.

https://user-images.githubusercontent.com/78004609/156289719-84a5515f-3bec-4cf5-ad46-7e424611b9e9.mp4

### Expected behavior 🤔

Make it possible to move the focus with the arrow keys even if the cell contains a Button.

### Steps to reproduce 🕹

Steps:

1. Click on the following link.
  https://codesandbox.io/s/rkwb6?file=/demo.tsx
2. Try to move cells with the arrow keys.

### Context 🔦

In the application we are developing, we are using buttons for specific DataGrid cells.
However, when a cell contains a button, it cannot be moved using the arrow keys.
This feature will improve the UX of people working by using the arrow keys to move around the cells.

### package.json

``` json
{
  "title": "RenderCellGrid Material Demo",
  "scripts": {
    "start": "react-scripts start"
  },
  "name": "RenderCellGrid Material Demo",
  "main": "index.tsx",
  "devDependencies": {
    "react-scripts": "latest"
  },
  "description": "https://github.com/mui/mui-x/blob/v5.5.1/docs/src/RenderCellGrid.tsx",
  "dependencies": {
    "typescript": "latest",
    "react-dom": "latest",
    "react": "latest",
    "@types/react-dom": "latest",
    "@types/react": "latest",
    "@mui/x-data-grid": "latest",
    "@mui/material": "latest",
    "@emotion/styled": "latest",
    "@emotion/react": "latest"
  }
}
```

### tsconfig

```json
{
  "compilerOptions": {
    "target": "es5",
    "lib": [
      "dom",
      "dom.iterable",
      "esnext"
    ],
    "allowJs": true,
    "skipLibCheck": true,
    "esModuleInterop": true,
    "allowSyntheticDefaultImports": true,
    "strict": true,
    "forceConsistentCasingInFileNames": true,
    "module": "esnext",
    "moduleResolution": "node",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "noEmit": true,
    "jsx": "react"
  },
  "include": [
    "src"
  ]
}
      
```

**Comment by m4theushw** [comment]
Already fixed by #3375

CodeSandbox: https://codesandbox.io/s/datagriddemo-material-demo-forked-7uhzdo?file=/demo.tsx

It will be included in the next release. Note that you need to use a small hack to manually stop the ripple of button.

**Comment by TaisonTsukada** [comment]
@m4theushw
I see.
Thanks for letting me know!
